gateway/ln-gateway/src/db.rs

```diff
@@ -23,6 +24,18 @@ use serde::{Deserialize, Serialize};
 use strum::IntoEnumIterator;
 use strum_macros::EnumIter;
 
+pub trait GatewayDbExt {
+    fn get_client_database(&self, federation_id: &FederationId) -> Database;
+}
+
+impl GatewayDbExt for Database {
+    fn get_client_database(&self, federation_id: &FederationId) -> Database {
+        let mut prefix = vec![DbKeyPrefix::ClientDatabase as u8];
+        prefix.append(&mut federation_id.consensus_encode_to_vec());
```

Comment:
in fedi we did incremented number (fed db prefix id) as prefix to get a small prefix and save lot of space (because this will repeated a lot)

then we store a map from federation id => db prefix id (u64)

ref https://github.com/fedixyz/fedi/blob/master/bridge/fedi-ffi/src/storage.rs#L512-L520

Reply:
> because this will repeated a lot

I have not verified, but I read that rocksdb impls diff encoding for keys, so any common prefixes are mostly free.

I still like the sequnce ids better. :)